### ISSUE TITLE
1. The translation length from 8-Point algorithm is always a unit (up-to-scale). So for the training stage, we align the predicted depth to the pose (triangulation depth) to make the depth and pose consistent for calculating the loss. For the inference stage of visual odometry, we need the scale consistency over the whole sequence thus we align the scale of pose translation to the predicted depth.

### ISSUE THREAD

**Issue by pleasegostraight** [B1ueber2y/TrianFlow#20]
1. The translation length from 8-Point algorithm is always a unit (up-to-scale). So for the training stage, we align the predicted depth to the pose (triangulation depth) to make the depth and pose consistent for calculating the loss. For the inference stage of visual odometry, we need the scale consistency over the whole sequence thus we align the scale of pose translation to the predicted depth.
2. Yes PWC-Net is originally a supervised optical flow method, but its architecture could be used for unsupervised optical flow training.

_Originally posted by @thuzhaowang in https://github.com/B1ueber2y/TrianFlow/issues/8#issuecomment-657198329_

**Comment by pleasegostraight** [comment]
And 8 points methods are not satisfied the scale consistency, so the pose is not scale consistency. Align the predict depth to the not scale consistented pose seems cannot solve the problem of depth consistency?

**Comment by scott89** [comment]
I have the same question regarding the temporal scale consistency. It seems that the only reasonable explanation for the predicted depth to be scale consistent is training with depth reprojection loss L_pd. However, it is still hard to interpret that the L_pd loss penalizes inconsistency between the scale-transformed depth D_a and D_b which are designed to be scale-invariant.

**Comment by SenZHANG-GitHub** [comment]
Yes it seems this paper only make sure the consistency between depth and pose at each time step separately while cannot account for the temporal consistency. The depth scale is achieved by aligning the predicted depth with depth from two-view triangulation, which assumes an unit translation for every frame-pair. Since the translation between two frames varies along the trajectory and the pose is aligned with the depth from triangulation, the unit translation assumption in triangulation leads to temporal inconsistency.
